types/artifacts.d.ts

```diff
@@ -23,6 +23,10 @@ declare global {
       fetchTime: string;
       /** A set of warnings about unexpected things encountered while loading and testing the page. */
       LighthouseRunWarnings: string[];
+      /** Whether the page was loaded on either a real or emulated mobile device. */
+      IsMobile: boolean;
+      /** Whether Lighthouse was run on a mobile device (i.e. not on a desktop machine). */
+      IsMobileHost: boolean;
```

Comment:
apparently, this is `false` if it was real mobile hardware but desktop was emulated. 
which isn't what I expected given the name.

Reply:
how about slightly different booleans?

`isMobileHardware`
`isMobileEmulated`

They will combine differently, but I think they communicate all the same things?

Reply:
> how about slightly different booleans?
> `isMobileHardware`
> `isMobileEmulated`

That kind of just punts the question back to the audits again, though. Really we just want to be able to easily answer "were my artifacts gathered in a mobile environment, whether real or emulated?".

The value should be true if run on desktop with mobile emulation, run on mobile with mobile emulation, or run on mobile with no emulation, otherwise false.

Maybe we should just drop `IsMobileHost` for now (I don't think anything needs it at this point?) and just pick a name we all like for `isMobile`/`IsMobileFormFactor`/`isMobileEmulated`?

Reply:
I like `isMobileHardware` to replace `isMobileHost` 👍 

when discussing @paulirish 's comment we just used the phrase `isMobileConditions` which I kinda like over `isMobile`/`isMobileFormFactor`. I'd want to avoid `Emulated` since that's not really what we are getting at

Reply:
`isMobileHardware`
`wasMobileConditions` 
???

the second is past tense because it's sometimes true and sometimes not. 
crazy but semantically reasonable?

Reply:
yeah, the `is` part is kind of weird since you're asking about how the data was collected, not the current state of things. OTOH we use `is` all over the place for exactly this kind of meaning and probably still should.

Could adapt to something like `isFromMobileConditions` to clear that up, though that's getting long winded :)

Reply:
I like `isMobileHardware`, but if we use that should `HostUserAgent` be `HardwareUserAgent`?

Maybe the user agent isn't even the right thing to look at. These are the special cases we have in mind right now:

- tap targets: is this device operated with imprecise inputs (i.e. touch)?
- font size: is this a small screen ([not 100% why that makes a difference](https://developers.google.com/web/tools/lighthouse/audits/font-sizes))
- viewport width: is this a small screen? I assume the reasoning for disabling it on desktop is that on desktop horizontal scrolling is more likely to be intentional, plus it's an SEO audit, not an accessibility one

A purely descriptive term for `isMobile` would be `isMobileNetworkUserAgent`, even though right now we use `HostUserAgent` and `emulatedFormFactor` to determine it.

Reply:
> apparently, this is `false` if it was real mobile hardware but desktop was emulated.
which isn't what I expected given the name.

Are you talking about `isMobileHost`? Right now I'd expect it to be true on a mobile device regardless of emulation.

Reply:
@mattzeunert Paul was just confused, he's got it now 👍 

I like `HardwareUserAgent` personally but that's a report-breaking change

Reply:
Renamed to `FormFactorIsMobile` and removed `IsMobileHost` from the base artifacts for now.

Reply:
at first read, I'd probably have to look up whether `FormFactorIsMobile` meant either tested on actual mobile hardware or the definition we're going for.

What about something around `TestedAsMobileDevice`?

Reply:
I'm starting to circle back around to @mattzeunert 's suggestion that this represents whether the device presented itself as mobile to the site, i.e network user agent was mobile :)

at this point though, I think we've all identified it's hard to name and we should just document well in the comment above it what it really represents

Reply:
user agent is independent of form factor, though, isn't it? That's why I was suggesting we go with whatever awkward string reasonably encapsulates `RanLighthouseWithADeviceThatEitherWasOrPretendedToBeAMobileOne`, which would include the UA string and screen characteristics

Reply:
`TestedAsMobileDevice` work for me, updated the PR.